src/ui/public/vislib/visualizations/point_series/column_chart.js

```diff
@@ -10,8 +10,11 @@ export function VislibVisualizationsColumnChartProvider(Private) {
     mode: 'normal',
     showTooltip: true,
     color: undefined,
-    fillColor: undefined
+    fillColor: undefined,
   };
+
+  const minBarWidth = 1;
```

Comment:
@ppisljar, I don't think there should be a limit.  I've split charts with 10 items.  On the dashboard, I just ensure that the panel size is large enough to show all of them.